pkg/engine/artifacts.go

```diff
@@ -33,120 +33,119 @@ func kubernetesContainerAddonSettingsInit(p *api.Properties) map[string]kubernet
+			destinationFile: reschedulerAddonDestinationFilename,
 		},
 		common.NVIDIADevicePluginAddonName: {
-			sourceFile:      "kubernetesmasteraddons-nvidia-device-plugin-daemonset.yaml",
+			sourceFile:      nvidiaAddonSourceFilename,
 			base64Data:      k.GetAddonScript(common.NVIDIADevicePluginAddonName),
-			destinationFile: "nvidia-device-plugin.yaml",
+			destinationFile: nvidiaAddonDestinationFilename,
 		},
 		common.ContainerMonitoringAddonName: {
-			sourceFile:      "kubernetesmasteraddons-omsagent-daemonset.yaml",
+			sourceFile:      containerMonitoringAddonSourceFilename,
 			base64Data:      k.GetAddonScript(common.ContainerMonitoringAddonName),
-			destinationFile: "omsagent-daemonset.yaml",
+			destinationFile: containerMonitoringAddonDestinationFilename,
 		},
 		common.IPMASQAgentAddonName: {
-			sourceFile:      "ip-masq-agent.yaml",
+			sourceFile:      ipMasqAgentAddonSourceFilename,
 			base64Data:      k.GetAddonScript(common.IPMASQAgentAddonName),
-			destinationFile: "ip-masq-agent.yaml",
+			destinationFile: ipMasqAgentAddonDestinationFilename,
 		},
 		common.AzureCNINetworkMonitorAddonName: {
-			sourceFile:      "azure-cni-networkmonitor.yaml",
+			sourceFile:      azureCNINetworkMonitorAddonSourceFilename,
 			base64Data:      k.GetAddonScript(common.AzureCNINetworkMonitorAddonName),
-			destinationFile: "azure-cni-networkmonitor.yaml",
+			destinationFile: azureCNINetworkMonitorAddonDestinationFilename,
 		},
 		common.DNSAutoscalerAddonName: {
-			sourceFile:      "dns-autoscaler.yaml",
+			sourceFile:      dnsAutoscalerAddonSourceFilename,
 			base64Data:      k.GetAddonScript(common.DNSAutoscalerAddonName),
-			destinationFile: "dns-autoscaler.yaml",
+			destinationFile: dnsAutoscalerAddonDestinationFilename,
 		},
 		common.CalicoAddonName: {
-			sourceFile:      "kubernetesmasteraddons-calico-daemonset.yaml",
+			sourceFile:      calicoAddonSourceFilename,
 			base64Data:      k.GetAddonScript(common.CalicoAddonName),
-			destinationFile: "calico-daemonset.yaml",
+			destinationFile: calicoAddonDestinationFilename,
 		},
 		common.AzureNetworkPolicyAddonName: {
-			sourceFile:      "kubernetesmasteraddons-azure-npm-daemonset.yaml",
+			sourceFile:      azureNetworkPolicyAddonSourceFilename,
 			base64Data:      k.GetAddonScript(common.AzureNetworkPolicyAddonName),
-			destinationFile: "azure-npm-daemonset.yaml",
+			destinationFile: azureNetworkPolicyAddonDestinationFilename,
 		},
 		common.AzurePolicyAddonName: {
-			sourceFile:      "azure-policy-deployment.yaml",
+			sourceFile:      azurePolicyAddonSourceFilename,
 			base64Data:      k.GetAddonScript(common.AzurePolicyAddonName),
-			destinationFile: "azure-policy-deployment.yaml",
+			destinationFile: azurePolicyAddonDestinationFilename,
 		},
 		common.CloudNodeManagerAddonName: {
-			sourceFile:      "kubernetesmasteraddons-cloud-node-manager.yaml",
+			sourceFile:      cloudNodeManagerAddonSourceFilename,
 			base64Data:      "",
```

Comment:
@mboersma do you have any idea why the cloud-node-manager addon doesn't currently support arbitrary override data like all the other user-configurable addons?

Reply:
There aren't any documented command-line arguments to the `cloud-node-manager` binary, so I left this empty when I implemented it.

Reply:
So, the `.Data` property of an addon is actually just meant to be base64-encoded data that represents the entire yaml manifest. It's a way of brute-force overriding what's in the blessed aks-engine implementation, either for testing, or just for the fun of living on the edge.

Because cloud-node-manager is just a yaml spec that we drop into /etc/kubernetes/addons/ I think there's no reason to specifically call it out as not able to use this brute-force override mechanism. Right?

Reply:
Gotcha. No, no reason that I can think of–it's just an oversight due to my misunderstanding.

Reply:
10-4, thanks for confirming! I'll go ahead and clean that up in this PR since it'll have the new UT coverage to prove the change out.